test/models/test_nn.py

```diff
@@ -305,26 +310,26 @@ def test_modes_DenseSymm(lattice, symmetries):
         bias_init=uniform(),
     )
 
-    dum_input = jax.random.normal(rng.next(), (3, 1, g.n_nodes))
+    dum_input = jax.random.normal(rng.next(), batch + (1, g.n_nodes))
 
     pars = ma_fft.init(rng.next(), dum_input)
     _ = ma_matrix.init(rng.next(), dum_input)
 
-    np.testing.assert_allclose(
-        ma_fft.apply(pars, dum_input), ma_matrix.apply(pars, dum_input)
-    )
+    out_fft = ma_fft.apply(pars, dum_input)
+    out_matrix = ma_matrix.apply(pars, dum_input)
+
+    assert out_fft.shape == tuple(batch + (4, len(perms)))
+    assert out_matrix.shape == tuple(batch + (4, len(perms)))
 
-    # Test Deprecation warning
-    dum_input_nofeatures = dum_input.reshape((dum_input.shape[0], dum_input.shape[2]))
+    np.testing.assert_allclose(out_fft, out_matrix)
+
+    # Test errors for less than 3 dimensions (missing features)
+    dum_input_nofeatures = dum_input.reshape((sum(batch), dum_input.shape[-1]))
```

Comment:
Yeah, I didn't realise what this test is doing... OTOH, that's not how dimensions work (even though 2*2=2+2 here)

```suggestion
    dum_input_nofeatures = dum_input.reshape((np.prod(batch), dum_input.shape[-1]))
```